app/assets/scripts/components/edit-map.js

```diff
@@ -60,16 +61,18 @@ const EditMap = React.createClass({
     this.map.addControl(this.drawPlugin);
     this.map.on('draw.create', () => this.handleDraw());
     this.map.on('draw.delete', () => this.handleDraw());
+    this.map.on('draw.update', () => this.handleDraw());
     this.startDrawing();
   },
 
-  loadExistingSource: function (prevAOI) {
-    prevAOI.id = 'edit-layer';
```

Comment:
By doing this you were modifying the original object, which leads to problems.
If you need to modify properties, you need to clone the object first.